bluesky_kafka/produce.py

```diff
@@ -0,0 +1,189 @@
+import logging
+
+import msgpack
+import msgpack_numpy as mpn
+
+
+# this is the recommended way to modify the python msgpack
+# package to handle numpy arrays with msgpack_numpy
+mpn.patch()
+
+logger = logging.getLogger(name="bluesky_kafka")
+
+
+def default_delivery_report(err, msg):
+    """
+    Called once for each message produced to indicate delivery result.
+    Triggered by poll() or flush().
+
+    Parameters
+    ----------
+    err : str
+    msg : Kafka message without headers
+    """
+
+    if err is not None:
+        logger.error("message delivery failed: %s", err)
+    else:
+        logger.debug(
+            "message delivered to topic %s [partition %s]",
+            msg.topic(),
+            msg.partition(),
+        )
+
+
+class BasicProducer:
+    """
+    Produce Kafka messages.
+
+    This class is intended for two purposes:
+        1) give bluesky users a simple way to produce general messages
+        2) provide a parent class for Publisher and future DocumentProducer
+
+    For guidance on flushing a BasicProducer see this:
+        https://github.com/confluentinc/confluent-kafka-python/issues/137
+
+    There is no default configuration. Consult Kafka documentation to determine
+    appropriate producer configuration for specific situations.
+
+    Parameters
+    ----------
+    topic : str
+        Topic to which all messages will be published.
+    bootstrap_servers: str
+        Comma-delimited list of Kafka server addresses as a string such as ``'127.0.0.1:9092'``.
+    key : str
+        Kafka "key" string. Specify any string to maintain message order. If None is specified
+        no ordering will be imposed on messages.
+    producer_config : dict, optional
+        Dictionary of configuration information used to construct the underlying
+        confluent_kafka.Producer.
+    on_delivery : function(err, msg), optional
+        A function to be called after a message has been delivered or after delivery has
+        permanently failed.
+    serializer : function, optional
+        Function to serialize data. Default is msgpack.dumps.
+
+    Example
+    -------
+
+    >>> producer = Producer(
```

Comment:
I think this should be `producer = BasicProducer(`

Reply:
right you are